### PR TITLE
Added integration tests to make sure retries happen for maven and ivy publication

### PR DESCRIPTION
**Pull request by rpalcolea**
Signed-off-by: Roberto Perez Alcolea <rperezalcolea@netflix.com>

### Context
Related to https://github.com/gradle/gradle/pull/12024

This PR adds tests to make sure maven and ivy publication can handle retires based on intermittent issues

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#git-commit---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [x] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [x] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by rpalcolea** [comment]
no problem @ljacomet! 

I'm starting to believe that https://github.com/gradle/gradle/blame/master/.travis.yml#L5 is causing trouble for PRs. Have you seen that in the past?  `Task :docs:docsTest FAILED`

File: subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishJavaRetriesIntegTest.groovy [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+        mavenRemoteRepo = new MavenHttpRepository(server, "/repo", mavenRepo)
+        module = mavenRemoteRepo.module("org.gradle.test", "testMavenRetries", "1.9")
+    }
+
+    def cleanup() {
+        server.stop()
+    }
+
+    def "should publish with intermittent network issues"() {
```
Comment (ljacomet) [inline subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishJavaRetriesIntegTest.groovy]: The test needs to be annotated with `@ToBeFixedForInstantExecution`

File: subprojects/ivy/src/integTest/groovy/org/gradle/api/publish/ivy/IvyPublishJavaRetriesIntegTest.groovy [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+        ivyRemoteRepo = new IvyHttpRepository(server, "/repo", ivyRepo)
+        module = ivyRemoteRepo.module("org.gradle.test", "testIvyRetries", "1.9")
+    }
+
+    def cleanup() {
+        server.stop()
+    }
+
+    def "should publish with intermittent network issues"() {
```
Comment (ljacomet) [inline subprojects/ivy/src/integTest/groovy/org/gradle/api/publish/ivy/IvyPublishJavaRetriesIntegTest.groovy]: The test needs to be annotated with `@ToBeFixedForInstantExecution`

Comment (rpalcolea) [reply subprojects/ivy/src/integTest/groovy/org/gradle/api/publish/ivy/IvyPublishJavaRetriesIntegTest.groovy]: https://github.com/gradle/gradle/pull/12127/commits/633967df88c13765d5fe60a5cda33dc898a739cc takes care of this

File: subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishJavaRetriesIntegTest.groovy (hunk 1 above)
Comment (rpalcolea) [reply subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishJavaRetriesIntegTest.groovy]: https://github.com/gradle/gradle/pull/12127/commits/633967df88c13765d5fe60a5cda33dc898a739cc

**Comment by rpalcolea** [comment]
@ljacomet are the issues related to the `- JAVA_TOOL_OPTIONS=-Dhttps.protocols=TLSv1.2` in Travis that changes outputs for `:docs:docsTest`?

**Comment by ljacomet** [comment]
Mostly trying to get tests to pass on our infra, as that's the complete check.
Though indeed the Travis CI error might need investigation, but should not be caused by your changes.

**Pull request merged**